### DISCUSSION TITLE
COMPILE_SIMULINK seems not be defined when using Visual Studio to rebuild the OpenFAST-Simulink

### DISCUSSION THREAD

**Discussion by zhzenith** [OpenFAST/openfast#1433] [Q&A]
Hi All,

I tried to rebuild FAST_SFunc for Simulink use after performing some modifications on HydroDyn. I then set  PCMode=4 to let the pitch control signal come from Simulink. However, there is the following error notation from MATLAB Command Window:

"AST_InitializeAll:SrvD_Init:ValidatePrimaryData:PCMode can equal 4 only when ServoDyn is
interfaced with Simulink or LabVIEW.  Set PCMode to 0, 3, or 5 or interface ServoDyn with
Simulink or LabVIEW."

During debugging, I found that the macro COMPILE_SIMULINK is not defined. Would you please help me to find some hints on solving the problem?

OpenFAST version: openfast-3.3.0
VS version: Visual Studio Community 2019, version 16.11.22

Thanks in advance.

Zhzenith

**Comment by zhzenith** [comment]
The problem was solved after I rebuild the solution. Thanks.

**Comment by GGBond007-tjw** [comment]
I had the same problem. Special attention should be paid to compiling in visual studio software. The correct property Settings should look like the following.

![F1](https://github.com/user-attachments/assets/ba44855c-f78a-4640-afea-f6e39503536c)

![F2](https://github.com/user-attachments/assets/5ce93c2b-8464-496b-9b13-1cf282b631c1)

Select the path to libmex.lib in the installation file for MATLAB.
![F3](https://github.com/user-attachments/assets/038d3f3f-128c-42df-98f1-a5724e64c3c6)

Select Release_MATLAB when compiling.
![image](https://github.com/user-attachments/assets/cc4b5081-f70a-4925-96d4-cd7dde672eed)

Then you can get the correct compilation result, so that no errors are reported at run time.
![image](https://github.com/user-attachments/assets/c1282e1a-5974-4aaa-93d2-82d0e1e266cc)
